lib/sudo-env/src/environment.rs

```diff
@@ -44,11 +44,19 @@ fn add_extra_env(context: &Context, environment: &mut Environment) {
         format!("{PATH_MAILDIR}/{}", context.target_user.name),
     );
     // The current SHELL variable should determine the shell to run when -s is passed, if none set use passwd entry
-    environment.insert("SHELL".to_string(), context.target_user.shell.clone());
+    // FIXME(pvdrz): this is lossy as not every `PathBuf` is a valid `String`.
+    environment.insert(
```

Comment:
Probably this lossy conversion can be fixed later by also using OSString's in the environment.